src/app/api/upload/route.ts

```diff
@@ -0,0 +1,32 @@
+import { NextRequest, NextResponse } from "next/server";
+import { ObjectCannedACL, PutObjectCommand } from "@aws-sdk/client-s3";
+import { v4 as uuidv4 } from "uuid";
+import { s3Client } from "@/app/api/upload/s3client";
+export const POST = async (req: NextRequest) => {
+    const data = await req.formData();
+    const file = data.get("file");
+
+    if (!(file instanceof File)) {
+        return NextResponse.json("No file");
+    }
+
+    const bytes = await file.arrayBuffer();
+    const buffer = Buffer.from(bytes);
+    const fileExtension = file.name.split(".").pop();
```

Comment:
Do not allow use custom extensions — it can be dangerous.
Fox example if someone upload an file like this 
```
hacker.throw\snew\sError/
```
see more https://security.stackexchange.com/questions/133347/does-changing-an-uploaded-executables-file-extension-to-png-render-it-safe

Reply:
У меня проверка типа файла происходить в момент загрузки в форму. Посмотри компонент UloadFile

Reply:
хакер через консоль или postman может послать файл

Reply:
Исправил. Добавил проверку типов файлов. Теперь на запрос через curl возвращается ответ со статусом 400